main.py

```diff
@@ -54,10 +70,72 @@ def configure_confluent_kafka_consumer(event, args):
         boto3.set_stream_logger()
         logger.debug(f"Using boto3 {boto3.__version__}")
 
-    logger.debug(event)
+        logger.debug(event)
+
+        message = json.loads(event['Records'][0]['Sns']['Message'])
+        logger.debug(message)
+
+        private_ip = message['detail']['containers'][0]['networkInterfaces'][0]['privateIpv4Address']
+        logger.debug(private_ip)
+
+    payload = {
+        "config": {
+            "connector.name": args.connector_name,
+            "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+            "tasks.max": args.tasks_max,
+            "topics": args.topics,
+            "flush.size": args.flush_size,
+            "s3.region": args.aws_region,
+            "s3.bucket.name": args.s3_bucket_name,
+            "storage.class": "io.confluent.connect.s3.storage.S3Storage",
+            "format.class": "io.confluent.connect.s3.format.avro.AvroFormat",
+            "schema.generator.class": "io.confluent.connect.storage.hive.schema.DefaultSchemaGenerator",
+            "partitioner.class": "io.confluent.connect.storage.partitioner.DefaultPartitioner",
+        }
+    }
+
+
+# Get a list of existing connectors
+
+    response = requests.get(f"http://{private_ip}:{args.port}/connectors")
+    existing_connector = response.content.decode("utf-8")
+    logger.debug(existing_connector)
+
+    # PUT payload if connectors do exist this updates existing connector
+
+    if existing_connector in existing_connector:
+        logger.debug("update connector [PUT]")
+
+        payload = {
+            "config": {
+                "connector.class": "io.confluent.connect.s3.S3SinkConnector",
+                "tasks.max": args.tasks_max,
+                "topics": args.topics,
+                "flush_size": args.flush_size,
+
+            }
+        }
+
+        response = requests.put(f"http://{private_ip}:{args.port}/connectors", json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+    else:
+        # POST payload if connectors don't exist
+
+        logger.debug("create connector [POST]")
+
+        response = requests.post(f"http://{private_ip}:{args.port}/connectors", json=payload)
+        logger.debug(response.content.decode("utf-8"))
+
+        # DELETE all others
+
+    for existing_connector in existing_connector:
+
+        if  existing_connector not in args.topics:
```

Comment:
I think you can remove this `if` statement.  Then outdent the `logger.debug()` call to align with the `response = ...` line (137).

Reply:
the if was needed, but it needed to be a different check (captured in new comment)